File: test/ch21-benchmarks/bench-009-json-parsing.jl
# BENCH-009: JSON parsing (50MB file) - Julia
# Parse large JSON and access deeply nested value

using JSON

function parse_and_access(filename::String)::String
    contents = read(filename, String)
    data = JSON.parse(contents)

    # Access deeply nested value
    city = data["users"][501]["profile"]["location"]["city"]  # Julia is 1-indexed
    return city
end

function main()
    filename = length(ARGS) > 0 ? ARGS[1] : "test-data/sample-50mb.json"
    city = parse_and_access(filename)
    # Silent for benchmarking
    # Expected: "London" or other city name
end

main()

File: test/ch21-benchmarks/bench-009-json-parsing.ts
// BENCH-009: JSON parsing (50MB file) - Deno TypeScript
// Parse large JSON and access deeply nested value

async function parseAndAccess(filename: string): Promise<string> {
    const contents = await Deno.readTextFile(filename);
    const data = JSON.parse(contents);

    // Access deeply nested value
    const city = data.users[500].profile.location.city;
    return city;
}

async function main(): Promise<void> {
    const filename = Deno.args[0] || "test-data/sample-50mb.json";
    const city = await parseAndAccess(filename);
    // Silent for benchmarking
    // Expected: "London" or other city name
}

main();
